### ISSUE TITLE
Updating CM and Secret via helm values does not roll the Deployments

### ISSUE THREAD

**Issue by domenicbove** [argoproj/argo-helm#2221]
### Describe the bug

When I update my helm values, and upgrade my helm deployment, the `argocd-secret` secret and `argocd-cm` config map get updated as I expect. But the argo application does not have hot reloading, so I've resorted to running
```
kubectl delete pods --all -n argocd
```
Which is probably overkill, but does work in getting the new properties read in. A better solution would be to add something like this: 
```
spec:
  template:
    metadata:
      annotations:
        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
```
To the deployment(s) that should get "rolled" when the CM/secret get updated.

See docs here: https://helm.sh/docs/howto/charts_tips_and_tricks/#automatically-roll-deployments

### Related helm chart

argo-cd

### Helm chart version

5.42.1

### To Reproduce

1. Deploy with some values
2. Update the values and run `helm upgrade`
3. Notice new values populate the CM/secret, but running apps dont honor

### Expected behavior

Only the required pods should roll

**Comment by mkilchhofer** [comment]
You are right, IMO the annotation for the checksum of the ConfigMap should be added on both:
- argocd-repo-server
- argocd-application-controller

But `argocd-server` will do hot-reload and does not need to be restarted via Helm.

Regarding your other case (the Secret): Why do we also need to restart on Secret change? AFAIK both controllers do not consume this secret.

**Comment by yu-croco** [comment]
I agree with @mkilchhofer and I also think we don't need to restart for `argocd-secret` Secret change, as far as I investigated in Argo CD (it wasn't referred from repo-server and application-controller). 🙋

**Comment by yu-croco** [comment]
Below is implemented in v5.45.4. Please reopen this issue again if there are missing . :)

> You are right, IMO the annotation for the checksum of the ConfigMap should be added on both:
>
> - argocd-repo-server
> - argocd-application-controller

**Comment by domenicbove** [comment]
Thanks for this addition! This chart is very well done! I've got my RBAC and Azure Auth all hooked up through values
